pkg/scheduler/core/generic_scheduler.go

```diff
@@ -678,7 +678,17 @@ func podFitsOnNode(
 		}
 	}
 
-	return len(failedPredicates) == 0, failedPredicates, nil
+	if len(failedPredicates) != 0 {
+		return false, failedPredicates, nil, nil
+	}
+
+	// Iterate each plugin to verify current node
+	status := filterPluginsFunc(pluginContext, pod, info.Node().Name)
+	if !status.IsSuccess() {
```

Comment:
This doesn't seem necessary. We can directly return, right?

Reply:
It would be more clear to seperate the good and base cases IMO

Reply:
Thanks @ahg-g review. IMO, we need to return a bool type value to indicate whether the current node matches.

Because the results cannot be directly determined based on failedPredicates or err, this is because [L663](https://github.com/kubernetes/kubernetes/blob/de1cfa9bc16a08116b8f6b56929d3b1ea39fd1e3/pkg/scheduler/core/generic_scheduler.go#L663) and [L681](https://github.com/kubernetes/kubernetes/blob/de1cfa9bc16a08116b8f6b56929d3b1ea39fd1e3/pkg/scheduler/core/Generic_scheduler.go#L681). Indeed, we can check both len(failedPredicates) and err, but this undoubtedly brings complexity to the judgment. We need more readable and easy to understand processing logic.